workflows/genomeassembly.nf

```diff
@@ -64,43 +56,42 @@ workflow GENOMEASSEMBLY {
     ch_versions = Channel.empty()
 
     //
-    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+    // SUBWORKFLOW: Read in yaml, validate and prepare for further steps
+    //   
+    PREPARE_INPUT(ch_input)
+    ch_versions = ch_versions.mix(PREPARE_INPUT.out.versions)
+
     //
-    INPUT_CHECK (
-        ch_input
-    )
-    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+    // Polishing step 1: map reads to the reference
+    //
+    PREPARE_INPUT.out.assemblies.map{ meta, p, h, merged -> [meta, merged] }.set{ fasta_merged_ch }
+    
+    LONGRANGER_MKREF(fasta_merged_ch)
+    ch_versions = ch_versions.mix(LONGRANGER_MKREF.out.versions)
+
+    PREPARE_INPUT.out.illumina_10X.map{ meta, reads, kmers -> [reads]}
+                    .set{ illumina_10X_ch }
+    LONGRANGER_ALIGN( LONGRANGER_MKREF.out.folder, illumina_10X_ch )
+    ch_versions = ch_versions.mix(LONGRANGER_ALIGN.out.versions)
 
     //
-    // MODULE: Run FastQC
+    // Polishing step 2: apply freebayes consensus based on longranger alignments
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
-    )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    LONGRANGER_ALIGN.out.bam.join(LONGRANGER_ALIGN.out.bai).set{ bam_ch }
+    PREPARE_INPUT.out.assemblies.join(PREPARE_INPUT.out.indices)
+                                .map{ meta, p, h, merged, p_i, h_i, merged_i -> [ merged, merged_i ] }
+                                .set{ reference_ch }
```

Comment:
Can lines 67 - 83 (everything to do with 10x data) go into a subworkflow instead?